photon-lib/src/main/java/org/photonvision/PhotonPoseEstimator.java

```diff
@@ -365,48 +364,44 @@ private Optional<EstimatedRobotPose> multiTagPNPStrategy(
             PhotonPipelineResult result,
             Optional<Matrix<N3, N3>> cameraMatrixOpt,
             Optional<Matrix<N5, N1>> distCoeffsOpt) {
+        // Arrays we need declared up front
+        var visCorners = new ArrayList<TargetCorner>();
+        var knownVisTags = new ArrayList<AprilTag>();
+
         if (result.getTargets().size() < 2) {
-            // Not enough targets -- run fallback strategy instead
             return update(result, cameraMatrixOpt, distCoeffsOpt, this.multiTagFallbackStrategy);
         }
 
         boolean hasCalibData = cameraMatrixOpt.isPresent() && distCoeffsOpt.isPresent();
-        if (!hasCalibData) {
-            // no calibration data -- run fallback strategy instead
-            return update(result, cameraMatrixOpt, distCoeffsOpt, this.multiTagFallbackStrategy);
-        }
-        Matrix<N3, N3> cameraMatrix = cameraMatrixOpt.get();
-        Matrix<N5, N1> distCoeffs = distCoeffsOpt.get();
 
-        ArrayList<TargetCorner> visCorners = new ArrayList<>();
-        ArrayList<AprilTag> knownVisTags = new ArrayList<>();
+        if(!hasCalibData) {
+            return Optional.empty();
```

Comment:
Needs to be `return update(result, cameraMatrixOpt, distCoeffsOpt, this.multiTagFallbackStrategy);`